src/wazuh_db/wdb_agent.c

```diff
@@ -11,268 +11,180 @@
+/* Update agent's last keepalive time. It opens and closes the DB. Returns 1 or -1 on error. */
 int wdb_update_agent_keepalive(int id, long keepalive) {
     int result = 0;
-    sqlite3_stmt *stmt;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_UPDATE_AGENT_KEEPALIVE, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_int64(stmt, 1, keepalive);
-    sqlite3_bind_int(stmt, 2, id);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_AGENT_KEEPALIVE], keepalive, id);
+    
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+    return result == 0 ? 1 : -1;
 }
 
 /* Delete agent. It opens and closes the DB. Returns 0 on success or -1 on error. */
 int wdb_remove_agent(int id) {
-    int result;
-    sqlite3_stmt *stmt;
-    char * name;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_DELETE_AGENT, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
+    int result = 0 ;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+    char * name = "";
 
     name = wdb_agent_name(id);
-
-    sqlite3_bind_int(stmt, 1, id);
-    result = wdb_step(stmt) == SQLITE_DONE;
-    sqlite3_finalize(stmt);
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_DELETE_AGENT], id);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput)) == 0;
     wdb_delete_agent_belongs(id);
 
     result = result && name ? wdb_remove_agent_db(id, name) : -1;
 
-    free(name);
+    os_free(name);
     return result;
 }
 
 /* Get name from agent. The string must be freed after using. Returns NULL on error. */
 char* wdb_agent_name(int id) {
-    sqlite3_stmt *stmt = NULL;
     char *result = NULL;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
-    if (wdb_open_global() < 0)
-        return NULL;
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_SELECT_AGENT], id);
 
-    if (wdb_prepare(wdb_global, SQL_SELECT_AGENT, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return NULL;
-    }
-
-    sqlite3_bind_int(stmt, 1, id);
-
-    switch (wdb_step(stmt)) {
-    case SQLITE_ROW:
-        result = strdup((char*)sqlite3_column_text(stmt, 0));
-        break;
-    case SQLITE_DONE:
-        result = NULL;
+    switch (wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput))) {
+    case 0:
+        os_strdup(wdboutput,result);
```

Comment:
Just one question. Does `wdboutput` contain just the name or is this a JSON with the name?

Reply:
This method had to be updated